<launch>
	<!-- Set this to your camera's name -->
  <arg name="cam_name" value="ardrone" />

  <!-- Start the GSCAM node -->
  <env name="GSCAM_CONFIG" value="v4l2src device=/dev/video0 ! video/x-raw-yuv,framerate=30/1,width=640,height=480 ! ffmpegcolorspace " />
  <!--node pkg="gscam" type="gscam" name="$(arg cam_name)">
   	<param name="camera_name" value="$(arg cam_name)" />
   	<param name="camera_info_url" value="/home/user/.ros/camera_info/${NAME}.yaml" />
  	<remap from="camera/image_raw" to="$(arg cam_name)/image_raw" />
  </node --> 

	<arg name="marker_size" default="9.36625" />
	<arg name="max_new_marker_error" default="0.08" />
	<arg name="max_track_error" default="0.2" />
	<arg name="cam_image_topic" default="/$(arg cam_name)/front/image_raw" />
	<arg name="cam_info_topic" default="/$(arg cam_name)/front/camera_info" />
	<arg name="output_frame" default="/ardrone_base_frontcam" />

	<node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg output_frame)" />

		<remap from="camera_image"  to="$(arg cam_image_topic)" />
		<remap from="camera_info"   to="$(arg cam_info_topic)" />
	</node>
</launch>

